cfme/tests/containers/test_overview_data_integrity.py

```diff
@@ -73,42 +66,14 @@ def test_containers_overview_data_integrity(provider):
     """
     navigate_to(ContainersOverview, 'All')
     # We should wait ~2 seconds for the StatusBox population
-    # (until we find a better solution)
-    # Since we collect images from Openshift and from the pods,
-    # images are tested separately
-
-    time.sleep(2)
-    statusbox_values = {data_set.object: int(StatusBox(data_set.name).value())
-                        for data_set in DATA_SETS}
-    api_values = get_api_object_counts(
-        list_providers_by_class(ContainersProvider))
-
-    list_img_from_registry = provider.mgmt.list_image()
-    list_img_from_registry_splitted = [i.id.split(
-        '@sha256:')[-1] for i in list_img_from_registry]
-
-    list_img_from_openshift = provider.mgmt.list_image_openshift()
-    list_img_from_openshift_splitted = [d['name']
-                                        for d in list_img_from_openshift]
-    list_img_from_openshift_parsed = [i[7:].split(
-        '@sha256:')[-1] for i in list_img_from_openshift_splitted]
-
-    list_img_from_registry_splitted_new = set(list_img_from_registry_splitted)
-    list_img_from_openshift_parsed_new = set(list_img_from_openshift_parsed)
-
-    list_img_from_openshift_parsed_new.update(list_img_from_registry_splitted_new)
-
-    num_img_cfme_56 = len(provider.mgmt.list_image())
-    num_img_cfme_57 = len(list_img_from_openshift_parsed_new)
-
-    assert StatusBox('images').value() == version.pick({version.LOWEST: num_img_cfme_56,
-                                                        '5.7': num_img_cfme_57})
+    wait_for(lambda: status_boxs[ContainersProvider].value(), num_sec=10, delay=1)
```

Comment:
**OPTIONAL**: Since I have now seen this pattern two times navigating to the same page, you could add the wait as part of the step to the navigation. I was tempted to make this required, but I think optional will suffice.